darwin/future/core/datasets/create_dataset.py

```diff
@@ -0,0 +1,37 @@
+from pydantic import parse_obj_as
+
+from darwin.future.core.client import Client
+from darwin.future.core.types import TeamSlug
+from darwin.future.data_objects.dataset import Dataset
+
+
+def create_dataset(api_client: Client, name: str) -> Dataset:
+    """
+    Creates a new dataset for the given team
+
+    Parameters
+    ----------
+
+    api_client: Client
+        The client to use to make the request
+    name: str
+        The name of the dataset to create
+
+    Returns
+    -------
+    Dataset
+        The created dataset
+
+    Raises
+    ------
+    HTTPError
+        Any HTTP errors returned by the API
```

Comment:
It'll also raise validation from pydantic if the returned HTTP objects aren't valid, not sure if we want to mention that

Reply:
Good point


> Check other PR for other changes might need to make.

Will do, these are only really "consilidation" PRs, an attempt to avoid the massive PRs, but tbh, I don't think the strategy is going well.  I need to come up with a better gitflow.